packages/moonstone/Scrollable/Scrollable.js

```diff
@@ -666,49 +666,93 @@ class ScrollableBase extends Component {
+			if (scrollLeft === limit) {
+				notifyAlreadyCompleted();
+				return false;
+			}
+		}
+		return true;
+	}
+
 	onVoice = (e) => {
 		const
 			scroll = e && e.detail && e.detail.scroll,
 			isRtl = this.uiRef.state.rtl;
-		this.isVoiceControl = true;
-
-		switch (scroll) {
-			case 'up':
-				this.voiceControlDirection = 'vertical';
-				this.onScrollbarButtonClick({isPreviousScrollButton: true, isVerticalScrollBar: true});
-				break;
-			case 'down':
-				this.voiceControlDirection = 'vertical';
-				this.onScrollbarButtonClick({isPreviousScrollButton: false, isVerticalScrollBar: true});
-				break;
-			case 'left':
-				this.voiceControlDirection = 'horizontal';
-				this.onScrollbarButtonClick({isPreviousScrollButton: !isRtl, isVerticalScrollBar: false});
-				break;
-			case 'right':
-				this.voiceControlDirection = 'horizontal';
-				this.onScrollbarButtonClick({isPreviousScrollButton: isRtl, isVerticalScrollBar: false});
-				break;
-			case 'top':
-				this.voiceControlDirection = 'vertical';
-				this.uiRef.scrollTo({align: 'top'});
-				break;
-			case 'bottom':
-				this.voiceControlDirection = 'vertical';
-				this.uiRef.scrollTo({align: 'bottom'});
-				break;
-			case 'leftmost':
-				this.voiceControlDirection = 'horizontal';
-				this.uiRef.scrollTo({align: isRtl ? 'right' : 'left'});
-				break;
-			case 'rightmost':
-				this.voiceControlDirection = 'horizontal';
-				this.uiRef.scrollTo({align: isRtl ? 'left' : 'right'});
-				break;
-			default:
-				this.isVoiceControl = false;
-		}
-		e.preventDefault();
+
+		if (this.canVoiceScroll(e)) {
+			this.isVoiceControl = true;
+			switch (scroll) {
+				case 'up':
+					this.voiceControlDirection = 'vertical';
+					this.onScrollbarButtonClick({isPreviousScrollButton: true, isVerticalScrollBar: true});
+					break;
+				case 'down':
+					this.voiceControlDirection = 'vertical';
+					this.onScrollbarButtonClick({isPreviousScrollButton: false, isVerticalScrollBar: true});
+					break;
+				case 'left':
+					this.voiceControlDirection = 'horizontal';
+					this.onScrollbarButtonClick({isPreviousScrollButton: !isRtl, isVerticalScrollBar: false});
+					break;
+				case 'right':
+					this.voiceControlDirection = 'horizontal';
+					this.onScrollbarButtonClick({isPreviousScrollButton: isRtl, isVerticalScrollBar: false});
+					break;
+				case 'top':
+					this.voiceControlDirection = 'vertical';
+					this.uiRef.scrollTo({align: 'top'});
+					break;
+				case 'bottom':
+					this.voiceControlDirection = 'vertical';
+					this.uiRef.scrollTo({align: 'bottom'});
+					break;
+				case 'leftmost':
+					this.voiceControlDirection = 'horizontal';
+					this.uiRef.scrollTo({align: isRtl ? 'right' : 'left'});
+					break;
+				case 'rightmost':
+					this.voiceControlDirection = 'horizontal';
+					this.uiRef.scrollTo({align: isRtl ? 'left' : 'right'});
+					break;
+				default:
+					this.isVoiceControl = false;
+			}
+			e.preventDefault();
+		}
```

Comment:
Basically, legacy code is something wrong because e.preventDefalut() has to called when voice control performed. So, I will change code for it will not called when default case of switch statement.